custom/plugins/VirtuaTechnology/Controllers/Frontend/Technologies.php

```diff
@@ -0,0 +1,161 @@
+<?php
+/**
+ * User: virtua
+ * Date: 2019-04-04
+ * Time: 15:55
+ *
+ * @author  Kuba Kułaga <intern4@wearevirtua.com>
+ * @link    https://github.com/virtIntern4a/kuba_shopware
+ */
+
+/**
+ * Class Technologies controller
+ */
+class Shopware_Controllers_Frontend_Technologies extends Enlight_Controller_Action
+{
+    /**
+     * Display paginated technologies with images
+     *
+     * @throws Exception
+     */
+    public function indexAction()
+    {
+        $sPage = intval($this->Request()->getParam('p', 1));
+        if ($sPage < 1) {
+            $sPage = 1;
+        }
+
+        $perSite = 2;
+
+        $paginator = $this->getTechnologiesPaginator($sPage, $perSite);
+        $technologies = $this->convertTechnologiesToArticleStruct(
+            $paginator->getIterator()->getArrayCopy()
+        );
+
+        /** @var \Shopware\Components\QueryAliasMapper $mapper */
+        $mapper = $this->get('query_alias_mapper');
+        $this->View()->assign([
+            'shortParameters' => $mapper->getQueryAliases(),
+            'showListing' => true,
+            'sPage' => $sPage,
+            'sArticles' => $technologies,
+            'theme' => ['infiniteScrolling' => false],
+        ]);
+
+        //todo pytanie: nie moglem przekaza zmiennej pages do widoku bo byla nadpisywana przez nie wiem co
+        //w koncu przekleilem kawalek action-pagination do technology/index.tpl i zmienilem nazwe zmiennej
+        //jak to zrobic lepiej?
```

Comment:
remove comment, this way is correct, you don't have to change the implementation